java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/LspTemplateUI.java

```diff
@@ -232,6 +211,18 @@ private LspTemplateUI() {
         });
     }
 
+    private static CompletionStage<FileBuilder> configure(FileBuilder builder, NbCodeLanguageClient client) {
+        CreateDescriptor desc = builder.createDescriptor(false);
+        FileObject template = desc.getTemplate();
+        Object handler = template.getAttribute(FileBuilder.ATTR_TEMPLATE_HANDLER);
+        if (handler == null) {
+            return client.showInputBox(new ShowInputBoxParams(Bundle.CTL_TemplateUI_SelectName(), desc.getProposedName())).thenApply(name -> {
+                return name != null ? builder.name(name) : null;
+            });
+        }
+        return CompletableFuture.completedFuture(builder);
```

Comment:
This is weird. If there is no `handler` we ask for a name. If there is some `handler` then we assume it will handle all the UI on itself. That's a change in the API. Handlers could have been written for NetBeans 13 for completely different reason.

Some other way to find out `hander` wants to provide additional customization is desirable.

Reply:
This could be IMHO solved as a separate PR. Once we agree upon how to provide an additional customization via `handler`, the implementation could be easily changed without breaking any API.